webservice/src/main/java/pl/edu/pjatk/lnpayments/webservice/payment/facade/PaymentFacade.java

```diff
@@ -73,11 +74,15 @@ public PaymentInfo buildInfoResponse(Optional<String> email) {
                 .build();
     }
 
-    public Payment createNewPayment(PaymentDetailsRequest paymentDetailsRequest, String email) {
+    public Payment createNewPayment(PaymentDetailsRequest paymentDetailsRequest) {
         int paymentExpiryInSeconds = propertyService.getPaymentExpiryInSeconds();
         int price = propertyService.getPrice();
         int numberOfTokens = paymentDetailsRequest.getNumberOfTokens();
-        User user = userService.findUserByEmail(email);
+        User user = null;
+        try {
+            user = userService.findUserByEmail(paymentDetailsRequest.getEmail());
```

Comment:
dlaczego coś takiego? Przecierz juz jest to obsługiwane w tej metodzie, nie musisz robić catcha

Reply:
jesli nie zrobie catcha to jest zwracane 404